### ISSUE TITLE
[Python2] AttributeError: 'list' object has no attribute 'stop'

### ISSUE THREAD

**Issue by pkolbus** [antlr/antlr4#1635]
Before submitting an issue to ANTLR, please check off these boxes:
 
- [X] I am not submitting a question on how to use ANTLR; instead, go to [antlr4-discussion google group](https://groups.google.com/forum/#!forum/antlr-discussion) or ask at [stackoverflow](http://stackoverflow.com/questions/tagged/antlr4)
- [X] I have done a search of the existing issues to make sure I'm not sending in a duplicate

On Ubuntu trusty (python2==2.7.6), the following Python exception occurs with antlr4-4.6-complete.jar (and corresponding python2 runtime from pip). With antlr4-4.5.3-complete.jar and antlr4-python2-runtime==4.5.2.1, there is no failure.

The script in question is in https://github.com/pkolbus/compiler-warnings

```
  File "/home/kolbus/compiler-warnings/parsers/parse-clang-diagnostic-groups.py", line 144, in <module>
    main(sys.argv)
  File "/home/kolbus/compiler-warnings/parsers/parse-clang-diagnostic-groups.py", line 126, in main
    tree = parser.expression()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 196, in expression
    self.statementExpression()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 246, in statementExpression
    self.switchDefinition()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 320, in switchDefinition
    self.diagnosticProperties()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 381, in diagnosticProperties
    self.classDefinition()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 533, in classDefinition
    self.switchName()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 664, in switchName
    self.switchText()
  File "/home/kolbus/compiler-warnings/parsers/TableGenParser.py", line 1069, in switchText
    self._errHandler.sync(self)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/error/ErrorStrategy.py", line 205, in sync
    if la==Token.EOF or la in recognizer.atn.nextTokens(s):
  File "/usr/local/lib/python2.7/dist-packages/antlr4/atn/ATN.py", line 61, in nextTokens
    return self.nextTokensNoContext(s)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/atn/ATN.py", line 55, in nextTokensNoContext
    s.nextTokenWithinRule = self.nextTokensInContext(s, None)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/atn/ATN.py", line 47, in nextTokensInContext
    return anal.LOOK(s, ctx=ctx)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/LL1Analyzer.py", line 74, in LOOK
    self._LOOK(s, stopState, lookContext, r, set(), set(), seeThruPreds, True)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/LL1Analyzer.py", line 162, in _LOOK
    self._LOOK(t.target, stopState, ctx, look, lookBusy, calledRuleStack, seeThruPreds, addEOF)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/LL1Analyzer.py", line 172, in _LOOK
    look.addSet(s)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/IntervalSet.py", line 88, in addSet
    self.addRange(i)
  File "/usr/local/lib/python2.7/dist-packages/antlr4/IntervalSet.py", line 68, in addRange
    if v.stop<i.start:
AttributeError: 'list' object has no attribute 'stop'
```
The 'list' is `[8, 9, 10, 11, 12, 13, 14, 15, 16, 17]` -- apparently an expansion of an interval, or simply a range().

Substituting the following for addSet avoids the error:
```
    def addSet(self, other):
        if other.intervals is not None:
            for i in other.intervals:
                if isinstance(i, Interval):
                    self.addRange(i)
                else:
                    for item in i:
                        self.addOne(item)
        return self
```

I'm happy to provide my workaround as a PR if t this is actually a reasonable fix. I'm also happy to bisect for the offending change if that would be useful.

**Comment by ericvergnaud** [comment]
Hi,

thanks for this.

An IntervalSet should only contain Intervals so your proposed fix would not fix the root cause i.e. a list which made its way though to addSet
Can you try and find out how this happens in your context?

Eric

**Comment by pkolbus** [comment]
Hi Eric,

Sure thing. I did a bit of git bisect & found the root cause; looks like a bad backport of a Python3 change, and a one line fix. Will have a PR up shortly.

-Peter

**Comment by pkolbus** [comment]
Hmmm, after rebasing and self-review, looks like the same change was already made to fix #1538. I missed that in my search because I was looking for the specific exception (AttributeError) and symbols.

**Comment by ericvergnaud** [comment]
Well it’s good to hear both that it is already fixed and that you tried to contribute.
Thanks
